### PR TITLE
Akshatpandya/api consistent

### LINKED ISSUE #301: Make API consistent
## Desired behavior

Some things that we should double-check:

* setters should start with `Set`
* POD arguments don't need to be declared `const`
* getters should be `const`
* see more on our [style guide](https://ignitionrobotics.org/docs/all/contributing#style-guides)

## Alternatives considered

Have users memorize what classes have what API styles.

## Implementation suggestion

New functions can be added to `ign-math6`, modifications need to target `main`. We've been deprecating the old functions on `main`, for example:

https://github.com/ignitionrobotics/ign-math/blob/931d058e7a9ffb8a123017d255fb876f6724630d/include/ignition/math/Quaternion.hh#L369-L378

But we could consider postponing these deprecation warnings, considering that they will impact lots of users. The goal is to provide users a nice consistent API, and we can accomplish this without adding migration burden.

### Here's the status:

1. [x] Angle
2. [ ] AxisAlignedBox
3. [ ] Box
4. [ ] Color
5. [ ] Cylinder
6. [ ] DiffDriveOdometry
7. [ ] Filter
8. [ ] Frustum
9. [x  GaussMarkovProcess 
10. [ ] graph/Edge
11. [ ] graph/GraphAlgorithms
12. [ ] graph/Graph
13. [ ] graph/Vertex
14. [ ] Helpers
15. [ ] Inertial
16. [ ] Kmeans
17. [ ] Line2
17. [ ] Line3
18. [ ] MassMatrix3
19. [ ] Material
20. [ ] MaterialType
21. [x] Matrix3
22. [ ] Matrix4
23. [ ] MovingWindowFilter
24. [ ] OrientedBox
25. [ ] PID
26. [ ] Plane
27. [ ] Pose3
28. [x] Quaternion. See [pull request #327](https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-math/pull-requests/327).
29. [ ] Rand
30. [ ] RollingMean
31. [ ] RotationSpline
32. [ ] SemanticVersion
33. [ ] SignalStats
34. [ ] Sphere
35. [ ] SphericalCoordiantes
36. [ ] Spline
37. [ ] Stopwatch
38. [ ] Temperature
39. [ ] Triangle3
40. [ ] Triangle
41. [ ]  Vector2
42. [ ] Vector3
43. [ ] Vector3Stats
44. [ ] Vector4

## Additional context

This was broken off #101.

### PR DESCRIPTION
**Pull request by akshatpandya**
# 🦟 Bug fix

Fixes #301 

## Summary
Making API consistent. Removing const from POD. Making sure getters are const. Other styling changes according to [style guide](https://google.github.io/styleguide/cppguide.html#Use_of_const).

Classes updated:
1. Angle
2. DiffDriveOdometry
3. Filter
4. Frustum
5. GaussMarkovProcess
6. graph/Edge
7. graph/GraphAlgorithms
8. graph/Graph
9. graph/Vertex

### CODE REVIEW

File: include/ignition/math/DiffDriveOdometry.hh [hunk 1]
```diff
@@ -129,8 +129,9 @@ namespace ignition
       /// \param[in] _wheelSeparation Distance between left and right wheels.
       /// \param[in] _leftWheelRadius Radius of the left wheel.
       /// \param[in] _rightWheelRadius Radius of the right wheel.
-      public: void SetWheelParams(double _wheelSeparation,
-                      double _leftWheelRadius, double _rightWheelRadius);
+      public: void SetWheelParams(const double _wheelSeparation,
+                      const double _leftWheelRadius,
+                      const double _rightWheelRadius);
```
Comment (chapulina) [inline include/ignition/math/DiffDriveOdometry.hh:134]: These should be left as `double` like you did on the other files

**Review by chapulina — COMMENTED** [review]
Thanks, this is great! I just have one comment about POF on `DiffDriveOdometry`

**Review by chapulina — APPROVED** [review]
Thanks!

**Comment by akshatpandya** [comment]
Hi @chapulina I've made the change you mentioned. I'll move onto the remaining classes now.

**Pull request merged**